src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2136,16 +2137,26 @@ internal void GenerateAnonymousFunctionConversionError(BindingDiagnosticBag diag
             // The simplest possible case is (x, y, z)=>whatever where the target type has a ref or out parameter.
 
             var delegateParameters = delegateType.DelegateParameters();
-            if (reason == LambdaConversionResult.RefInImplicitlyTypedLambda)
+            if (reason == LambdaConversionResult.MismatchedParameterRefKind)
             {
                 for (int i = 0; i < anonymousFunction.ParameterCount; ++i)
                 {
                     var delegateRefKind = delegateParameters[i].RefKind;
-                    if (delegateRefKind != RefKind.None)
+                    var lambdaRefKind = anonymousFunction.RefKind(i);
+
+                    if (lambdaRefKind != delegateRefKind)
```

Comment:
Shouldn't we be using `OverloadResolution.AreRefsCompatibleForMethodConversion` instead of `!=` here?

Reply:
yup!